azure-storage-common/src/Common/Internal/Authentication/OAuthAuthScheme.php

```diff
@@ -0,0 +1,251 @@
+ * @link      https://github.com/azure/azure-storage-php
+ */
+
+namespace MicrosoftAzure\Storage\Common\Internal\Authentication;
+
+use GuzzleHttp\Psr7\Request;
+use MicrosoftAzure\Storage\Common\Internal\Http\HttpFormatter;
+use MicrosoftAzure\Storage\Common\Internal\Resources;
+use MicrosoftAzure\Storage\Common\Internal\Utilities;
+
+/**
+ * Provides shared key authentication scheme for blob and queue. For more info
+ * check: http://msdn.microsoft.com/en-us/library/windowsazure/dd179428.aspx
+ *
+ * @ignore
+ * @category  Microsoft
+ * @package   MicrosoftAzure\Storage\Common\Internal\Authentication
+ * @author    Azure Storage PHP SDK <dmsh@microsoft.com>
+ * @copyright 2019 Microsoft Corporation
+ * @license   https://github.com/azure/azure-storage-php/LICENSE
+ * @link      https://github.com/azure/azure-storage-php
+ */
+class OAuthAuthScheme implements IAuthScheme
+{
+
+    /**
+     * The Bearer Token used
+     */
+    protected $bearerToken;
+    /**
+     * The included headers
+     */
+    protected $includedHeaders;
+
+
+    /*
+    *Auth header 
+    */
+
+
+    /**
+     * Constructor.
+     *
+     * @param string $bearerToken
+     * 
+     * @return OAuthAuthScheme
+     */
+    
+    public function __construct($bearerToken)
+    {
+        $this->bearerToken  = $bearerToken;
+
+        $this->includedHeaders   = array();
+        $this->includedHeaders[] = Resources::CONTENT_ENCODING;
+        $this->includedHeaders[] = Resources::CONTENT_LANGUAGE;
+        $this->includedHeaders[] = Resources::CONTENT_LENGTH;
+        $this->includedHeaders[] = Resources::CONTENT_MD5;
+        $this->includedHeaders[] = Resources::CONTENT_TYPE;
+        $this->includedHeaders[] = Resources::DATE;
+        $this->includedHeaders[] = Resources::IF_MODIFIED_SINCE;
+        $this->includedHeaders[] = Resources::IF_MATCH;
+        $this->includedHeaders[] = Resources::IF_NONE_MATCH;
+        $this->includedHeaders[] = Resources::IF_UNMODIFIED_SINCE;
+        $this->includedHeaders[] = Resources::RANGE;
+    }
+
+    /**
+     * Returns authorization header to be included in the request.
+     *
+     
+     *
+     * @see Specifying the Authorization Header section at
+     *      http://msdn.microsoft.com/en-us/library/windowsazure/dd179428.aspx
+     *
+     * @return string
+     */
+    public function getAuthorizationHeader()
+    {
+        return $this->bearerToken;
+        
+    }
+
+    /**
+     * Computes canonicalized headers for headers array.
+     *
+     * @param array $headers request headers.
+     *
+     * @see Constructing the Canonicalized Headers String section at
+     *      http://msdn.microsoft.com/en-us/library/windowsazure/dd179428.aspx
+     *
+     * @return array
+     */
+    protected function computeCanonicalizedHeaders($headers)
+    {
+        $canonicalizedHeaders = array();
+        $normalizedHeaders    = array();
+        $validPrefix          =  Resources::X_MS_HEADER_PREFIX;
```

Comment:
Minor: extra space after assignment operator.

Reply:
these code is not needed for OAuth